toolchain/check/type_structure.h

```diff
@@ -77,24 +77,40 @@ class TypeStructure : public Printable<TypeStructure> {
  private:
   friend class TypeStructureBuilder;
 
+  // Elements of the type structure, indicating the present of a concrete or
+  // symbolic element, and for aggregate concrete types (such as generic types),
+  // nesting for the types inside.
   enum class Structural : uint8_t {
+    // A concrete element in the type structure, such as `bool`.
     Concrete,
+    // A concrete element in the type structure that contains nested types
+    // within, such as `C(D)` for some classes C and D. It marks the start of
+    // the nested and is paired with a ConcreteCloseParen at the end of the
+    // nested types.
     ConcreteOpenParen,
+    // Closes a ConcreteOpenParen for a concrete type with nested types.
```

Comment:
Suggesting for whitespace, and sentence to emphasize the type relationship -- do as you will with this.
```suggestion
    //
    // Closes a ConcreteOpenParen for a concrete type with nested types.
    // Does not have its own concrete type.
```